api/v1beta1/placementapi_webhook.go

```diff
@@ -126,19 +127,33 @@ func (r *PlacementAPI) ValidateDelete() (admission.Warnings, error) {
 }
 
 func (r PlacementAPISpec) ValidateCreate(basePath *field.Path) field.ErrorList {
-	return ValidateDefaultConfigOverwrite(basePath, r.DefaultConfigOverwrite)
+	return r.PlacementAPISpecCore.ValidateCreate(basePath)
 }
 
 func (r PlacementAPISpec) ValidateUpdate(old PlacementAPISpec, basePath *field.Path) field.ErrorList {
-	return ValidateDefaultConfigOverwrite(basePath, r.DefaultConfigOverwrite)
+	return r.PlacementAPISpecCore.ValidateCreate(basePath)
 }
 
 func (r PlacementAPISpecCore) ValidateCreate(basePath *field.Path) field.ErrorList {
-	return ValidateDefaultConfigOverwrite(basePath, r.DefaultConfigOverwrite)
+	var allErrs field.ErrorList
+
+	// validate the service override key is valid
+	allErrs = append(allErrs, service.ValidateRoutedOverrides(basePath, r.Override.Service)...)
+
+	allErrs = append(allErrs, ValidateDefaultConfigOverwrite(basePath, r.DefaultConfigOverwrite)...)
+
+	return allErrs
 }
 
 func (r PlacementAPISpecCore) ValidateUpdate(old PlacementAPISpecCore, basePath *field.Path) field.ErrorList {
-	return ValidateDefaultConfigOverwrite(basePath, r.DefaultConfigOverwrite)
+	var allErrs field.ErrorList
+
+	// validate the service override key is valid
+	allErrs = append(allErrs, service.ValidateRoutedOverrides(basePath, r.Override.Service)...)
```

Comment:
I'm wondering if ValidateRoutedOverrides could be more generic if the caller pass a path that is already in sync with the field value passed in. E.g. if in a specific service operator the override service field is somewhere else in the instance then the following patter can keep the lib-common code ignorant about that:
```golang
	allErrs = append(allErrs, service.ValidateRoutedOverrides(basePath.Child("override").Child("service"), r.Override.Service)...)
```

Reply:
done